server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5198,6 +5199,35 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
             }
         }
 
+        List<Long> networkIds = cmd.getNetworkIds();
+        if (ImageFormat.OVA.equals(template.getFormat())) {
+            List<NetworkPrerequisiteTO> networkPrerequisiteTOList =
+                    templateDetailsDao.listNetworkRequirementsByTemplateId(template.getId());
+            Map <Integer, Long> networkMap = cmd.getVmNetworkMap();
+            if (CollectionUtils.isNotEmpty(networkPrerequisiteTOList)) {
+                networkIds = new ArrayList<>();
+                Network defaultNetwork = null;
+                if (zone.isSecurityGroupEnabled()) {
+                    defaultNetwork = _networkModel.getNetworkWithSGWithFreeIPs(zone.getId());
+                    if (defaultNetwork == null) {
+                        throw new InvalidParameterValueException("No network with security enabled is found in zone ID: " + zone.getUuid());
+                    }
+                } else {
+                    defaultNetwork = getDefaultNetwork(zone, owner);
+                    if (defaultNetwork == null) {
+                        throw new InvalidParameterValueException(String.format("Default network not found for zone ID: %s and account ID: %s", zone.getUuid(), owner.getUuid()));
+                    }
+                }
+                for (NetworkPrerequisiteTO networkPrerequisiteTO : networkPrerequisiteTOList) {
+                    Long networkId = networkMap.get(networkPrerequisiteTO.getInstanceID());
+                    if (networkId == null && networkPrerequisiteTO.isAutomaticAllocation()) {
+                        networkId = defaultNetwork.getId();
+                    }
+                    networkIds.add(networkId);
+                }
+            }
+        }
+
```

Comment:
@DaanHoogland this is what I proposed. API will supply a list of networkIds with same size as the number of NICs in the template. Though some elements in this list can be null when user has not passed any mapping and automaticAllocation is false.

Now the problem with this is, at some point during deploy we check if the network/network id in the list is not null. That will need some refactoring. Also, refactoring will be needed while creating NicProfiles https://github.com/shapeblue/cloudstack/blob/599b43a7ca08f55e6becb991ad0541647f318373/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java#L3630-L3705